src/apps/explorer/components/OrderWidget/OrderDetails.tsx

```diff
@@ -38,57 +48,108 @@ const Table = styled(SimpleTable)`
   }
 `
 
-// TODO: use tooltip component once we have tooltips
-const questionMark = <FontAwesomeIcon icon={faQuestionCircle} />
-
 // TODO: either use a RichOrder object or transform it here
 // TODO: for that we'll need token info (decimals, symbol)
-export type Props = { order: RawOrder }
+export type Props = { order: RawOrder; buyToken: TokenErc20; sellToken: TokenErc20 }
 
 export function OrderDetails(props: Props): JSX.Element {
-  const { order } = props
-  const { uid } = order
+  const { order, buyToken, sellToken } = props
+  const { uid, owner, kind, partiallyFillable, creationDate, validTo, buyAmount, sellAmount, executedFeeAmount } = order
+
+  const status = useMemo(() => getOrderStatus(order), [order])
+
+  const [invertedPrice, setInvertedPrice] = useState(false)
+  const invertPrice = useCallback(() => setInvertedPrice((curr) => !curr), [])
+
+  const limitPrice = useMemo(
+    () =>
+      getOrderLimitPrice({
+        order,
+        buyTokenDecimals: buyToken.decimals,
+        sellTokenDecimals: sellToken.decimals,
+        inverted: invertedPrice,
+      }),
+    [buyToken.decimals, invertedPrice, order, sellToken.decimals],
+  )
+
+  const { amount: filledAmount, percentage: filledPercentage } = useMemo(() => getOrderFilledAmount(order), [order])
+  const { executedBuyAmount, executedSellAmount } = useMemo(() => getOrderExecutedAmounts(order), [order])
 
   return (
     <Table
       body={
         <>
           <tr>
-            <td>{questionMark}Order Id</td>
+            <td>Order Id</td>
             <td>{uid}</td>
           </tr>
           <tr>
-            <td>{questionMark}Sell order</td>
-            <td>Swap 10 WETH for DAI</td>
+            <td>From</td>
+            <td>{owner}</td>
```

Comment:
I'd add a link here. Maybe I'll just jump to it after the review, cause I plan to merge this beauty of a PR :)